### PR TITLE
Adding 'num_columns_recent' to config.

### PR DESCRIPTION
**Pull request by michelle192837**
This is useful for tracking a few values in TestGrid, like stale tests
or some soon-to-come summary features.

### CODE REVIEW

**Comment by krzyzacy** [comment]
I think you'll need to add it to https://github.com/michelle192837/test-infra/blob/26b883e03aae66a4fd49751f19d65897adc6ae59/testgrid/config/yaml2proto/yaml2proto.go#L43-L64

maybe add a test as well.

/ok-to-test

**Comment by krzyzacy** [comment]
oh, plz commit the new config.pb.go :-)

File: testgrid/config/yaml2proto/yaml2proto.go [hunk 1]
```diff
@@ -73,6 +77,10 @@ func ReconcileDashboardtab(currentTab *config.DashboardTab, defaultTab *config.D
 		currentTab.CodeSearchPath = defaultTab.CodeSearchPath
 	}
 
+  if currentTab.NumColumnsRecent == 0 {
+    currentTab.NumColumnsRecent = defaultTab.NumColumnsRecent
+  }
```
Comment (krzyzacy) [inline testgrid/config/yaml2proto/yaml2proto.go]: :smiling_imp: one more space

Comment (michelle192837) [inline testgrid/config/yaml2proto/yaml2proto.go]: Fixed...?

Comment (michelle192837) [inline testgrid/config/yaml2proto/yaml2proto.go]: Nope, now fixed.

File: testgrid/config/config.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ default_test_group:
   column_header:
   - configuration_value: Commit # Shows the commit number on column header
   - configuration_value: infra-commit
+  num_columns_recent: 10
```
Comment (krzyzacy) [inline testgrid/config/config.yaml]: does this mean it's only going to show 10 most recent results now?

Comment (michelle192837) [inline testgrid/config/config.yaml]: Nope; it'll mean that for logic that cares about a number of recent columns (for instance, how many columns back to look to see if a tab is flaky), we'll be able to calculate that.

**Comment by krzyzacy** [comment]
Consider also mention it's used for stuff like flake and summary?
Maybe add a section about this option in the [readme.md](https://github.com/kubernetes/test-infra/blob/master/testgrid/config/README.md)

**Comment by krzyzacy** [comment]
merging and I'll update config

**Comment by michelle192837** [comment]
I'll be updating readme.md with other config stuff as well as this. ^^

**Pull request merged**